src/scheduler/server_info.cpp

```diff
@@ -974,104 +973,104 @@ free_server_psets(std::unordered_map<std::string, node_partition *>& spsets)
 
 /**
  * @brief	dup a sinfo->svr_to_psets map (deep copy)
- * 			Note: this might not be needed once we convert node_partition to a class
+ *			Note: this might not be needed once we convert node_partition to a class
  *
  * @param[in]	spsets - map of server psets
- * @param[in]	sinfo - sinfo in the duplicated universe
  *
- * @return std::unordered_map<std::string, node_partition *>
- * @retval An unordered_map containing copy of the input
+ * @return nothing
  */
-static std::unordered_map<std::string, node_partition *>
-dup_server_psets(std::unordered_map<std::string, node_partition *>& spsets, server_info *sinfo)
+void
+server_info::dup_server_psets(const std::unordered_map<std::string, node_partition*>& spsets)
 {
-	std::unordered_map<std::string, node_partition *> newpset;
-
-	for (const auto& spset : spsets) {
-		newpset[spset.first] = dup_node_partition(spset.second, sinfo);
-		if (newpset[spset.first] == NULL) {
-			free_server_psets(newpset);
-			return {};
-		}
-	}
 
-	return newpset;
+	for (const auto& spset : spsets)
+		svr_to_psets[spset.first] = dup_node_partition(spset.second, this);
```

Comment:
Can dup_node_partition() no longer return NULL?

Reply:
Sorry, I forgot to respond. dup_node_partition() can return NULL in some cases.